lib/led/rgb.js

```diff
@@ -172,12 +172,22 @@ class RGB {
       intensity: 100,
       isAnode: options.isAnode || false,
       interval: null,
+      // isRunning is used to tracke whether an Animation is in progress
+      // Copying pattern used in led.js.
+      isRunning: false,
       // red, green, and blue store the raw color set via .color()
       // values takes state into account, such as on/off and intensity
       values: {
         red: 255,
         green: 255,
         blue: 255,
+      },
+      // state.prev records the last color set using color(),
+      // and is used to determine the new color when calling on() or pulse()
+      prev: {
```

Comment:
`state.prev` was added in prior implementation as well. @islemaster adeptly explains the reasoning for this addition [here](https://github.com/code-dot-org/johnny-five/pull/10/files#r160806071).